lib/adapters/idb/index.js

```diff
@@ -816,10 +816,14 @@ function init(api, opts, callback) {
 
   var req;
   if (opts.storage) {
-    req = indexedDB.open(dbName, {
-      version: ADAPTER_VERSION,
-      storage: opts.storage
-    });
+    try { // option only available in Firefox 26+
+      req = indexedDB.open(dbName, {
+        version: ADAPTER_VERSION,
+        storage: opts.storage
+      });
+    } catch(err) {
+        req = indexedDB.open(dbName, ADAPTER_VERSION);
+    }
```

Comment:
Nice! Can you please move this try/catch to a separate function to avoid deoptimizations in V8? Try/catches [deoptimize their entire parent function](https://github.com/petkaantonov/bluebird/wiki/Optimization-killers).